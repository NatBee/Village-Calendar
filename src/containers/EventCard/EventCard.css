.add-event {
  width: 300px;
  display: inline-block;
  background-color: #A7B9AB;
  opacity: .85;
  padding: 50px 40px 36px 40px;
  margin: 60px; }

.add-event form {
  display: flex;
  flex-direction: column;
  align-items: center; }

h2 {
  font-size: 1.25em;
  padding: 0 30px;
  margin: 0 0 45px; }

h3 {
  margin: 0 0 20px; }

.add-event button {
  background-color: #3E4D4E;
  height: 30px;
  color: #DBE3D4;
  font-family: 'Exo', sans-serif;
  border: 2px solid #293435;
  margin: 0 0 5px;
  width: 70px; }

.add-event button:hover {
  background-color: #293435; }

.add-event form input {
  margin: 10px 0 0px;
  font-size: 1em;
  padding: 2px 6px;
  width: 275px; }

.add-event button {
  width: 100px; }

.add-event form .cancel-btn {
  margin: 25px 45px 0 0px; }

.add-event form .add-event-btn {
  margin: 25px 0px 0 45px; }
