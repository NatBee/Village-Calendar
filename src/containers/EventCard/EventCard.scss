.add-event {
  width: 300px;
  display: inline-block;
  background-color: #A7B9AB;
  opacity: .85;
  padding: 50px 40px 36px 40px;
  margin: 60px;
}

.add-event form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

h2 {
  font-size: 1.25em;
  padding: 0 30px;
  margin: 0 0 45px;
}

h3 {
  margin: 0 0 20px;
}

.btn {
  display: flex;
  justify-content: space-around;
}

.add-event button { 
  background-color: #3E4D4E;
  height: 30px;
  color: #DBE3D4;
  font-family: 'Exo', sans-serif;
  border: 2px solid #293435;
  margin: 0 0 5px;
  width: 70px;
}

.add-event button:hover {
  background-color: #293435;
}

.add-event form input {
  margin: 10px 0 0px;
  font-size: 1em;
  padding: 2px 6px;
  width: 275px;
}

.add-event button {
  width: 100px;
  margin: 0px 0 6px;
}

.add-event form button {
  margin: 20px 0 0;
}

.form-btn {
  display: flex;
}