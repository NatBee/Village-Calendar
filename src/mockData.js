export const mockData = {
  user: {
    user: 'Dk', 
    credential: 'Xg', 
    additionalUserInfo: 'Pg', 
    operationType: "signIn"
  },
  events: [
    {
      start: "2018-02-09T09:00:00-07:00",
      end: "2018-02-09T10:00:00-07:00",
      title: "Super Fun Party",
      description: "Fun with friends",
      eventID: "7js6e24fo0kin4qcotoegu37hg",
      location: 'Mi Casa'
    }
  ],
  token: 'token_number',
  calendarID: 'calendarID',
  village: [{natalie: 'nataliesmontoya@gmail.com'}],
  time: {
    startTime: "2018-02-28T19:00:00.000Z", 
    endTime: "2018-02-28T20:00:00.000Z"
  },
  event: {
    start: "2018-02-09T09:00:00-07:00",
    end: "2018-02-09T10:00:00-07:00",
    title: "Super Fun Party",
    description: "Fun with friends",
    eventID: "7js6e24fo0kin4qcotoegu37hg",
    location: 'Mi Casa'
  }
}