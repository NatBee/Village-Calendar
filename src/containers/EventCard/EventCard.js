import React, { Component } from 'react';
import './EventCard.css';
import { addEventToCalendar } from '../../helper/apiCall';
import { connect } from 'react-redux';

class EventCard extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      summary: '',
      location: ''
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.name;
    const value = e.target.value;
    this.setState({ [field]: value });
  }

  addEvent = async (e) => {
    await addEventToCalendar(this.props.calendarID, this.props.time, this.state);
  }

  render() {
    return (
      <div>
        <h1>Event</h1>
        <form onSubmit={this.addEvent}>
          <input 
            type='text' 
            name='title' 
            placeholder='Event title' 
            onChange={this.handleChange} 
          />
          <input 
            type='text' 
            name='location' 
            placeholder='Event location' 
            onChange={this.handleChange} 
          />
          <input 
            type='text' 
            name='summary' 
            placeholder='Event Description' 
            onChange={this.handleChange} 
          />
          <button>Submit</button>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  time: store.time,
  calendarID: store.calendarID,
  user: store.user
})

export default connect(mapStateToProps)(EventCard);