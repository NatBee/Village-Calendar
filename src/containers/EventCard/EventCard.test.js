/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { EventCard, mapStateToProps } from './EventCard';
import { mockData } from '../../mockData';

describe('EventCard', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<EventCard event={mockData.event}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should map the store correctly', () => {
    const mockStore = {
      time: mockData.time,
      calendarID: mockData.calendarID,
      user: mockData.user,
      token: mockData.token  
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  })

  it('should have a default state', () => {
    const wrapper = shallow(<EventCard event={mockData.event}/>);
    const expected = {
      title: '',
      summary: '',
      location: ''
    }
    expect(wrapper.state()).toEqual(expected);
  })
})