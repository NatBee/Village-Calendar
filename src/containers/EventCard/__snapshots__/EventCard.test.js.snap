// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    event={
      Object {
        "description": "Fun with friends",
        "end": "2018-02-09T10:00:00-07:00",
        "eventID": "7js6e24fo0kin4qcotoegu37hg",
        "location": "Mi Casa",
        "start": "2018-02-09T09:00:00-07:00",
        "title": "Super Fun Party",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Event
        </h1>,
        <form
          onSubmit={[Function]}
        >
          <input
            name="title"
            onChange={[Function]}
            placeholder="Event title"
            type="text"
          />
          <input
            name="location"
            onChange={[Function]}
            placeholder="Event location"
            type="text"
          />
          <input
            name="summary"
            onChange={[Function]}
            placeholder="Event Description"
            type="text"
          />
          <button>
            Submit
          </button>
        </form>,
        <button
          onClick={[Function]}
        >
          Cancel
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Event",
        },
        "ref": null,
        "rendered": "Event",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="title"
              onChange={[Function]}
              placeholder="Event title"
              type="text"
            />,
            <input
              name="location"
              onChange={[Function]}
              placeholder="Event location"
              type="text"
            />,
            <input
              name="summary"
              onChange={[Function]}
              placeholder="Event Description"
              type="text"
            />,
            <button>
              Submit
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "title",
              "onChange": [Function],
              "placeholder": "Event title",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "location",
              "onChange": [Function],
              "placeholder": "Event location",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "summary",
              "onChange": [Function],
              "placeholder": "Event Description",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Event
          </h1>,
          <form
            onSubmit={[Function]}
          >
            <input
              name="title"
              onChange={[Function]}
              placeholder="Event title"
              type="text"
            />
            <input
              name="location"
              onChange={[Function]}
              placeholder="Event location"
              type="text"
            />
            <input
              name="summary"
              onChange={[Function]}
              placeholder="Event Description"
              type="text"
            />
            <button>
              Submit
            </button>
          </form>,
          <button
            onClick={[Function]}
          >
            Cancel
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Event",
          },
          "ref": null,
          "rendered": "Event",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="title"
                onChange={[Function]}
                placeholder="Event title"
                type="text"
              />,
              <input
                name="location"
                onChange={[Function]}
                placeholder="Event location"
                type="text"
              />,
              <input
                name="summary"
                onChange={[Function]}
                placeholder="Event Description"
                type="text"
              />,
              <button>
                Submit
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "title",
                "onChange": [Function],
                "placeholder": "Event title",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "location",
                "onChange": [Function],
                "placeholder": "Event location",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "summary",
                "onChange": [Function],
                "placeholder": "Event Description",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
